### PR TITLE
Added a field for kafka header timestamp in DatastreamProducerRecord

### PR DESCRIPTION
**Pull request by jzakaryan**
Important: DO NOT REPORT SECURITY ISSUES DIRECTLY ON GITHUB.  
For reporting security issues and contributing security fixes,  
please, email security@linkedin.com instead, as described in  
the contribution guidelines.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
   // timestamp of when the record was sent to transport provider
   private Optional<Long> _eventsSendTimestamp = Optional.empty();
 
+  private Optional<Long> _eventsAuditTimestamp = Optional.empty();
```
Comment (somandal) [inline datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java:34]: Can you add a comment explaining this field comes from the Kafka header? Also my mistake, we probably shouldn't have the word "audit" in the name. How about renaming it to: `_eventsKafkaHeaderTimestamp`?

File: datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java [hunk 2]
```diff
@@ -93,6 +95,20 @@ public void setEventsSendTimestamp(long timestamp) {
     _eventsSendTimestamp = Optional.of(timestamp);
   }
 
+  /**
+   * Get timestamp in Epoch-millis for when the events have been audited
```
Comment (somandal) [inline datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java:99]: nit: can you rephrase this:
Get saved Kafka header timestamp in Epoch-millis 
or something along those lines?

File: datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java [hunk 3]
```diff
@@ -93,6 +95,20 @@ public void setEventsSendTimestamp(long timestamp) {
...
+   */
+  public Optional<Long> getEventsAuditTimestamp() {
+    return _eventsAuditTimestamp;
+  }
+
+  /**
+   * Set timestamp in Epoch-millis for when the events have been audited
```
Comment (somandal) [inline datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java:106]: nit: can you rephrase this:
Set the timestamp from the Kafka header in Epoch-millis 
or something along those lines?

File: datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java [hunk 4]
```diff
@@ -149,4 +165,5 @@ public String getCheckpoint() {
   public Optional<String> getDestination() {
     return _destination;
   }
+
```
Comment (somandal) [inline datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java:168]: nit: remove extra space

File: datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java (hunk 1 above)
Comment (jzakaryan) [reply datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java:34]: I was kind of reluctant to add the word audit here too. I guess you're right. Will try to rephrase everything in more generic terms.
